indexer/services/comlink/src/controllers/api/v4/vault-controller.ts

```diff
@@ -0,0 +1,168 @@
+import { stats } from '@dydxprotocol-indexer/base';
+import {
+  DEFAULT_POSTGRES_OPTIONS,
+  Ordering, PaginationFromDatabase,
+  PnlTicksFromDatabase,
+  PnlTicksTable,
+  QueryableField,
+  perpetualMarketRefresher,
+  PerpetualMarketFromDatabase,
+} from '@dydxprotocol-indexer/postgres';
+import express from 'express';
+import {
+  Controller, Get, Route,
+} from 'tsoa';
+import _ from 'lodash';
+
+import { getReqRateLimiter } from '../../../caches/rate-limiters';
+import config from '../../../config';
+import { aggregatePnlTicks, handleControllerError } from '../../../lib/helpers';
+import { rateLimiterMiddleware } from '../../../lib/rate-limit';
+import ExportResponseCodeStats from '../../../request-helpers/export-response-code-stats';
+import { pnlTicksToResponseObject } from '../../../request-helpers/request-transformer';
+import { MegavaultHistoricalPnlResponse, VaultsHistoricalPnlResponse, VaultHistoricalPnl } from '../../../types';
+
+const router: express.Router = express.Router();
+const controllerName: string = 'vault-controller';
+
+// TODO(TRA-570): Placeholder interface for mapping of vault subaccounts to tickers until vaults
+// table is added.
+interface VaultMapping {
+  [subaccountId: string]: string;
+}
+
+@Route('vault/v1')
+class VaultController extends Controller {
+  @Get('/megavault/historicalPnl')
+  async getMegavaultHistoricalPnl(): Promise<MegavaultHistoricalPnlResponse> {
+    const vaultPnlTicks: PnlTicksFromDatabase[] = await getVaultSubaccountPnlTicks();
+
+    // aggregate pnlTicks for all vault subaccounts grouped by blockHeight
+    const aggregatedPnlTicks: Map<number, PnlTicksFromDatabase> = aggregatePnlTicks(vaultPnlTicks);
+
+    return {
+      megavaultsPnl: Array.from(aggregatedPnlTicks.values()).map(
+        (pnlTick: PnlTicksFromDatabase) => {
+          return pnlTicksToResponseObject(pnlTick);
+        }),
+    };
+  }
+
+  @Get('/v1/vaults/historicalPnl')
```

Comment:
Actually do you even need to put the "v1" here since its already on the route above?

Reply:
Will add a comment, I put a v1 since this is an MVP and there's corners being cut that we may want to rectify in the future and seemed better to have versioning already added rather than add it in later.
As for `vault` vs `vaults`, agree it looks redundant, but `vault` is the top-level while `vaults` indicates these are not the `megavault` specific data.

Reply:
> Also, are we now trying to add versioning to our indexer APIs?

Experimenting here since there might need to be a new version of this API for other vault data, and there's more strict type checking on FE/mobile on both protocol and indexer responses and a need for backwards compatability.